### DISCUSSION TITLE
Eloquent Collection dehydration with validation rules

### DISCUSSION THREAD

**Discussion by juliomotol** [livewire/livewire#4107] [7. Bugs (archived)]
Livewire has a problem dehydrating Eloquent Collections that has a defined rule. Given the following:

```php
class PostsComponent extends Component
{
    public EloquentCollection $posts;

    public function rules()
    {
        return [
            'posts' => ['required', 'min:2'],
            'posts.*.title' => ['required'],
            'posts.*.content' => ['required'],
        ]
    }
}
```

it dehydrates the `$posts` into:

```json
{
    ...
    "posts": {
        "posts": null,
        "0": {
            "title": "Foo",
            "content": "Foo Content"
        },
        "1": {
            "title": "Bar",
            "content": "Bar Content"
        }
    }
    ...
}
```

Now the real problem starts. Initial responses works fine, but when sending back to the server, its says `Livewire encountered corrupt data when trying to hydrate the … component` since in the js side it "sorts" the `posts`' property, so when it tries to message back to the server will be comparing the ff data with the checksum:

```json
{
    ...
    "posts": {
        "0": {
            "title": "Foo",
            "content": "Foo Content"
        },
        "1": {
            "title": "Bar",
            "content": "Bar Content"
        },
        "posts": null // this was initially the first item but was sent to the back since it was sorted alphabetically
    }
    ...
}
```

and breaks the whole operation.

When having the `'posts' => ['required', 'min:2']` rule removed, it then works fine as it should.

**Comment by joshhanley** [comment]
@juliomotol thanks for reporting! I wrote the deep nesting stuff, and have a fair idea what might be causing this. Will have a look when I get the chance.

Thanks!

**Reply by juliomotol** [reply]
Should i create and issue for this?

**Reply by joshhanley** [reply]
@juliomotol you have already 🙂

This is the issue as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/livewire/livewire/discussions/3147
